### PR TITLE
Use m_vertexAttrib instead of deprecated attributes

### PR DESCRIPTION
**Pull request by Ventona666**
_Check if you branch history is PR compatible_
- Your branch need to be up to date with origin/master AND to have linear history (i.e. no merge commit).
- Update your git repository `git fetch origin` if origin is this remote
- Check with the script provided in `scripts/is-history-pr-compatible.sh`
- You must use clang-format style
_These checks are enforced by github workflow actions_
_Please refer to the corresponding log in case of failure_

_UPDATE the form below to describe your PR._

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Moving all deprecated attributes (m_normal, m_tangent, m_bitangent, m_texCoord) and m_vertex to m_vertexAttribs (which is an attribManager).

* **What is the current behavior?** (You can also link to an open issue here)
File loaders use deprecated attributes to load data.

* **What is the new behavior (if this is a feature change)?**
Now file loaders use an attribManager (m_vertexAttrib) to store this data instead of deprecated attributes, all tests passed.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
There is no breaking change.

### CODE REVIEW

**Comment by nmellado** [comment]
Hum, weird, there is a commit incrementing Radium version. what happened ?

**Comment by dlyr** [comment]
I propose to review the PR as is ... and I'll clean the history.

**Comment by nmellado** [comment]
yep ok.
It's mentioned WIP in the title, so let's wait until it is ready.

File: src/Core/Asset/GeometryData.hpp [hunk 1]
```diff
@@ -214,21 +234,74 @@ class RA_CORE_API GeometryData : public AssetData
...
+     *
+     * @tparam Container
+     * @param name
+     * @return Get container base on name (lock).
+     * @warning If AttribHandle corresponding to name doesn't exist, it's created
+     * and return. By using this method, user has read-write access to data, data is lock, when
+     * done call attribDataUnlock( std :: std::string name )
+     */
+    template <typename Container>
+    inline Container& getAttribDataWithLock( std::string name );
```
Comment (nmellado) [inline src/Core/Asset/GeometryData.hpp:266]: Consider 
```suggestion
    inline Container& getAttribDataWithLock( const std::string& name );
```
instead (same for all the other methods).

Comment (nmellado) [reply src/Core/Asset/GeometryData.hpp]: thx

**Comment by Ventona666** [comment]
I must also modify libGltf plugin by adding attribDataUnlock() where is needed.

**Comment by MathiasPaulin** [comment]
> In this proposal, several functionalities of the Attrib manager are replicated (e.g. access to an attrib from its name). I am not sure if it is a good idea to to that, as the loaders will have, _in fine_, to play with the attribute manager directly.
> 
> What do you think @dlyr @MathiasPaulin ?

I set the "WIP" tag as this PR is not yet ready to review. If you want to review, wait until the WIP tag is removed and the tag "Ready to review" is set.

I recall you that we already discussed (IRL) about splitting  the work @Ventona666 do into two PRs. So we do. 
  - This PR only removes deprecated attribs in GeometryData by implementing all the interfaces using the attribManager then mark these interfaces as deprecated and proposes a much more simpler wrapper over attribArrayGeometry as an alternative. Then, sure, it keeps (except only a few renaming) the same (redundant) API than before but depreciate this API.
  - A second PR will remove all these deprecated (and redundant) methods by using directly the wrapper on a MutliArrayGeometry.
 
Right now, this is WIP and subject to numerous changes before the "Ready To Review" tag will be set.

**Comment by nmellado** [comment]
Please convert to draft if you don’t want reviews.

**Pull request closed without merging**